default.go

```diff
@@ -109,7 +113,20 @@ func IsHostname(str string) bool {
 			valid = false
 		}
 	}
-	return valid
+
+	if !valid {
+		return false
+	}
+
+	if len(parts) > 1 && strings.ContainsRune(parts[len(parts)-1], '-') {
+		// if the top-level domain name contains hyphens, then it must be a valid puny code representation.
+		// We use idna to further check the TLD part
+		_, err := idna.Registration.ToUnicode(strings.ToLower(parts[len(parts)-1]))
+
+		return err == nil
+	}
+
+	return true
```

Comment:
I would like to suggest something else


```suggestion
	if len(parts) == 0 {
		return false
	}

	tld := parts[len(parts)-1]
	if !strings.Contains(tld, "-") {
		return true
	}
	
	// the top-level domain name contains hyphens, then it must be a valid puny code representation.
	// We use idna to further check the TLD part
	_, err := idna.Registration.ToUnicode(strings.ToLower(tld))
	return err == nil
```

Reply:
Yes I get your point; BTW this inspires me another check, since there are some invalid hosts in the test suite that are considered valid